src/client_side_request.h

```diff
@@ -82,155 +80,149 @@ class ClientHttpRequest
-    /// neither the current request nor the parsed request URI are known
-    void setErrorUri(const char *errorUri);
-
-    /// Prepares to satisfy a Range request with a generated HTTP 206 response.
-    /// Initializes range_iter state to allow raw range_iter access.
-    /// \returns Content-Length value for the future response; never negative
-    int64_t prepPartialResponseGeneration();
-
-    /// Build an error reply. For use with the callouts.
-    void calloutsError(const err_type error, const ErrorDetail::Pointer &errDetail);
-
-    /// if necessary, stores new error information (if any)
-    void updateError(const Error &error);
-
-#if USE_ADAPTATION
-    // AsyncJob virtual methods
-    virtual bool doneAll() const {
-        return Initiator::doneAll() &&
-               BodyConsumer::doneAll() && false;
-    }
-    virtual void callException(const std::exception &ex);
-#endif
+    ClientRequestContext *calloutContext = nullptr;
 
 private:
     /// assigns log_uri with aUri without copying the entire C-string
-    void absorbLogUri(char *aUri);
+    void absorbLogUri(char *);
     /// resets the current request and log_uri to nil
     void clearRequest();
     /// initializes the current unassigned request to the virgin request
     /// sets the current request, asserting that it was unset
-    void assignRequest(HttpRequest *aRequest);
+    void assignRequest(HttpRequest *);
 
-    int64_t maxReplyBodySize_;
-    StoreEntry *entry_;
-    StoreEntry *loggingEntry_;
-    ConnStateData * conn_;
+    int64_t maxReplyBodySize_ = 0;
+    StoreEntry *entry_ = nullptr;
+    StoreEntry *loggingEntry_ = nullptr;
+    ConnStateData * conn_ = nullptr;
 
 #if USE_OPENSSL
-    /// whether (and how) the request needs to be bumped
-    Ssl::BumpMode sslBumpNeed_;
-
 public:
     /// returns raw sslBump mode value
     Ssl::BumpMode sslBumpNeed() const { return sslBumpNeed_; }
     /// returns true if and only if the request needs to be bumped
     bool sslBumpNeeded() const { return sslBumpNeed_ == Ssl::bumpServerFirst || sslBumpNeed_ == Ssl::bumpClientFirst || sslBumpNeed_ == Ssl::bumpBump || sslBumpNeed_ == Ssl::bumpPeek || sslBumpNeed_ == Ssl::bumpStare; }
     /// set the sslBumpNeeded state
-    void sslBumpNeed(Ssl::BumpMode mode);
+    void sslBumpNeed(Ssl::BumpMode);
     void sslBumpStart();
-    void sslBumpEstablish(Comm::Flag errflag);
+    void sslBumpEstablish(Comm::Flag);
+
+private:
+    /// whether (and how) the request needs to be bumped
+    Ssl::BumpMode sslBumpNeed_ = Ssl::bumpEnd;
 #endif
 
 #if USE_ADAPTATION
-
 public:
-    void startAdaptation(const Adaptation::ServiceGroupPointer &g);
+    void startAdaptation(const Adaptation::ServiceGroupPointer &);
     bool requestSatisfactionMode() const { return request_satisfaction_mode; }
 
+    /* AsyncJob API */
+    virtual bool doneAll() const override {
+        return Initiator::doneAll() &&
+               BodyConsumer::doneAll() &&
+               false; // XXX: why?
```

Comment:
This class has not been converted into a proper AsyncJob yet. Among many other (far more important) things, the class does not express the job termination condition correctly. Full refactoring is a difficult task because it requires, among other things, removing synchronous access to this class public methods.

```suggestion
               false; // TODO: Refactor into a proper AsyncJob
```